agent/catalog_endpoint.go

```diff
@@ -31,23 +41,24 @@ func (s *HTTPServer) CatalogRegister(resp http.ResponseWriter, req *http.Request
 	// Forward to the servers
 	var out struct{}
 	if err := s.agent.RPC("Catalog.Register", &args, &out); err != nil {
-		metrics.IncrCounterWithLabels([]string{"client", "rpc", "error", "catalog_register"}, 1,
-			[]metrics.Label{{Name: "node", Value: s.nodeName()}})
+		labels = append(labels, metrics.Label{Name: "code", Value: "500"})
+		metrics.IncrCounterWithLabels([]string{"client", "rpc", "error", "catalog_register"}, 1, labels)
 		return nil, err
 	}
-	metrics.IncrCounterWithLabels([]string{"client", "api", "success", "catalog_register"}, 1,
-		[]metrics.Label{{Name: "node", Value: s.nodeName()}})
+	labels = append(labels, metrics.Label{Name: "code", Value: "200"})
+	metrics.IncrCounterWithLabels([]string{"client", "api", "success", "catalog_register"}, 1, labels)
 	return true, nil
 }
 
 func (s *HTTPServer) CatalogDeregister(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
-	metrics.IncrCounterWithLabels([]string{"client", "api", "catalog_deregister"}, 1,
-		[]metrics.Label{{Name: "node", Value: s.nodeName()}})
+	labels := []metrics.Label{{Name: "agent", Value: s.nodeName()}, {Name: "client", Value: findSourceIP(req)}}
```

Comment:
Hmm I'm not sure if that is something we'd want to do. In Prometheus at least every distinct label combination is a different series. Having something with such a high and potentially unbounded cardinality as the _client_ IP in HTTP handler seems like a bad thing in such a general place.

It might be something people who know their clients are well behaved could opt in to but just on by default seems like it could really blow up the number of series Consul is generating.

Even splitting all these metrics by agent host might be a big deal for some of our users who have thousands of nodes in their cluster and will suddenly have many thousand times more distinct series after this rolls out.